attributes/attributes.go

```diff
@@ -112,30 +112,31 @@ func (a *Attributes) String() string {
 	sb.WriteString("{")
 	first := true
 	for k, v := range a.m {
-		var key, val string
-		if str, ok := k.(interface{ String() string }); ok {
-			key = str.String()
-		} else if str, ok := k.(string); ok {
-			key = str
-		}
-		if str, ok := v.(interface{ String() string }); ok {
-			val = str.String()
-		} else if str, ok := v.(string); ok {
-			val = str
-		}
 		if !first {
 			sb.WriteString(", ")
 		}
-		sb.WriteString(fmt.Sprintf("%q: %q ", key, val))
+		sb.WriteString(fmt.Sprintf("%s: %s ", str(k), str(v)))
 		first = false
 	}
 	sb.WriteString("}")
 	return sb.String()
 }
 
+func str(x interface{}) string {
+	if v, ok := x.(fmt.Stringer); ok {
+		return fmt.Sprintf("%q", v.String())
+	} else if v, ok := x.(string); ok {
+		return fmt.Sprintf("%q", v)
+	}
+	return fmt.Sprintf("<%p>", x)
```

Comment:
This one doesn't print literal quotes.

Why not keep it the way it was?  Have this return the actual string and quote it when it's used?

Reply:
Yeah, makes sense.